### ISSUE TITLE
contenttype name with _ (underscore) not allowed

### ISSUE THREAD

**Issue by jadwigo** [bolt/bolt#4674]
If you create a content type with an underscore in the identifier the database update will break. The following will result in a lot of notices like `Notice: Undefined index: custom_things in /[]/html/src/Storage/Database/Schema/Builder/ContentTables.php on line 33`

```yml
custom_things:
  name: Things
  singular_name: Thing
  fields:
    title:
      type: text
      required: true
      label: "Title"
    slug:
      type: slug
      uses: title
    body:
      type: textarea
      label: "Body"
```

Also the fields `title` and `body` will not be created, so the database update will keep nagging you that the database is broken.

**Comment by GawainLynch** [comment]
@bobdenotter wasn't there a reason for this? One of the database platforms?

**Comment by jadwigo** [comment]
in the urls in the back-end it will be changed to http://example.com/bolt/overview/custom-things .. so it also might have something to do with the slugify

**Comment by GawainLynch** [comment]
It should be separate, it's handled [here](https://github.com/bolt/bolt/blob/master/src/Config.php#L485-L489)… but I only fixed it a couple of weeks ago, so depending on how old your checkout of master is…

**Comment by GawainLynch** [comment]
c.f. PR #4578 and original issue #4567

**Comment by GawainLynch** [comment]
Fixed in #4578
